vscode-dotnet-sdk-extension/src/extension.ts

```diff
@@ -0,0 +1,167 @@
+    });
+    const versionResolver = new VersionResolver(context.globalState, eventStream);
+
+    const dotnetAcquireRegistration = vscode.commands.registerCommand(`${commandPrefix}.${commandKeys.acquire}`, async (commandContext: IDotnetAcquireContext) => {
+        const pathResult = callWithErrorHandling(async () => {
+            eventStream.post(new DotnetSDKAcquisitionStarted());
+
+            let version: string | undefined = commandContext ? commandContext.version : undefined;
+            if (!version) {
+                version = await vscode.window.showInputBox({
+                    placeHolder: '5.0',
+                    value: '5.0',
+                    prompt: '.NET version, i.e. 5.0',
+                });
+            }
+            if (!version) {
+                displayWorker.showErrorMessage('No .NET SDK version provided', () => { /* No callback needed */ });
+                return undefined;
+            }
+
+            eventStream.post(new DotnetAcquisitionRequested(version!));
+            const resolvedVersion = await versionResolver.getFullSDKVersion(version!);
+            const dotnetPath = await acquisitionWorker.acquireSDK(resolvedVersion);
+            displayWorker.showInformationMessage(`.NET SDK ${version} installed to ${dotnetPath.dotnetPath}`, () => { /* No callback needed */ });
+            setPathEnvVar(path.dirname(dotnetPath.dotnetPath));
+            return dotnetPath;
+        }, issueContext(undefined, 'acquireSDK'));
+        return pathResult;
+    });
+    const dotnetUninstallAllRegistration = vscode.commands.registerCommand(`${commandPrefix}.${commandKeys.uninstallAll}`, async (commandContext: IDotnetUninstallContext | undefined) => {
+        await callWithErrorHandling(async () => {
+            await acquisitionWorker.uninstallAll();
+            displayWorker.showInformationMessage('All VS Code copies of the .NET SDK uninstalled.', () => { /* No callback needed */ });
+        }, issueContext(commandContext ? commandContext.errorConfiguration : undefined, 'uninstallAll'));
+    });
+    const showOutputChannelRegistration = vscode.commands.registerCommand(`${commandPrefix}.${commandKeys.showAcquisitionLog}`, () => outputChannel.show(/* preserveFocus */ false));
+    const reportIssueRegistration = vscode.commands.registerCommand(`${commandPrefix}.${commandKeys.reportIssue}`, async () => {
+        const [url, issueBody] = formatIssueUrl(undefined, issueContext(AcquireErrorConfiguration.DisableErrorPopups, 'reportIssue'));
+        await vscode.env.clipboard.writeText(issueBody);
+        open(url);
+    });
+
+    context.subscriptions.push(
+        dotnetAcquireRegistration,
+        dotnetUninstallAllRegistration,
+        showOutputChannelRegistration,
+        reportIssueRegistration);
+    context.subscriptions.push({
+        dispose: () => vscode.Disposable.from(...eventStreamObservers).dispose(),
```

Comment:
I don't think you need to wrap the disposable in another disposable, you can just push it along side the other registrations 😄